mobile/src/components/scan/CustomBottomSheet.jsx

```diff
@@ -1,8 +1,9 @@
 import React, {useCallback, useMemo, useRef, useState} from 'react';
```

Comment:
this is a pretty complex function and it would be very helpful to add docstring at the top of the function describing what the function does and how it does it. You can also add comments inside the function to clarify stuff.

Reply:
Added comments and docstring with ca93d7c0869db922b8b393bb8392c321abf2d17c